pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/AggregateOperator.java

```diff
@@ -160,17 +164,17 @@ private void consumeInputBlocks() {
         for (int rowId = 0; rowId < numRows; rowId++) {
           Object[] row = SelectionOperatorUtils.extractRowFromDataTable(dataBlock, rowId);
           Key key = extraRowKey(row, _groupSet);
-          int keyHashCode = key.hashCode();
-          _groupByKeyHolder.put(keyHashCode, key.getValues());
-          for (int i = 0; i < _aggregationFunctions.length; i++) {
-            Object currentRes = _groupByResultHolders[i].get(keyHashCode);
+          _groupByKeyHolder.put(key, key.getValues());
+          for (int i = 0; i < _aggCalls.size(); i++) {
+            Object currentRes = _groupByResultHolders[i].get(key);
+            // TODO: fix that single agg result (original type) has different type from multiple agg results (double).
```

Comment:
isn't this already addressed in #9676?

Reply:
This is one was written on top that since I want to share the test util.  Let me rebase after the merge succeeds.